api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -112,38 +111,47 @@ public void writeFile(String bucketName, String fileName, byte[] bytes) {
     storage.create(blobInfo, bytes);
   }
 
+  private JSONObject getCredentialsBucketJSON(final String objectPath) {
+    return new JSONObject(readCredentialsBucketString(objectPath));
+  }
+
   @Override
   public JSONObject getJiraCredentials() {
-    return new JSONObject(readToString(getCredentialsBucketName(), "jira-login.json"));
+    return getCredentialsBucketJSON("jira-login.json");
   }
 
   @Override
   public JSONObject getElasticCredentials() {
-    return new JSONObject(readToString(getCredentialsBucketName(), "elastic-cloud.json"));
+    return getCredentialsBucketJSON("elastic-cloud.json");
+  }
+
+  private String readCredentialsBucketString(String objectPath) {
+    return readToString(getCredentialsBucketName(), objectPath);
+  }
+
+  private GoogleCredential getBucketCredential(final String objectPath) throws IOException {
+    String json = readCredentialsBucketString(objectPath);
+    return GoogleCredential.fromStream(new ByteArrayInputStream(json.getBytes()));
   }
 
   @Override
   public GoogleCredential getGSuiteAdminCredentials() throws IOException {
-    String json = readToString(getCredentialsBucketName(), "gsuite-admin-sa.json");
-    return GoogleCredential.fromStream(new ByteArrayInputStream(json.getBytes()));
+    return getBucketCredential("gsuite-admin-sa.json");
```

Comment:
This is a big improvement.

The only other thing I'd do is to put these constant JSON file names into a .properties file that you could swap out per environment if needed. This will make the Java code more easily portable.